### PR TITLE
Current Price Method, Trade Objects Step Attribute, Window Size Integrated with Obs Space, Talib Transformer Fixed, Basic Rendering Functionality, Base Dash Files

### PR DESCRIPTION
**Pull request by RaedShabbir**
A variety of changes across the repo, loosely outlined below. I will be cleaning this up a bit more in the next few days. 

trade.py 
- trade objects now have a step attribute to avoid confusion in regards to the correct step for other calculations 

trading_environment.py
- self._current_step is now updated at the end of the self._step() and set during self._reset() 
- rendering functionality ported in and modified from RLTrader 
- self._info has been modified to work with callbacks in stable baselines, hence the 'episode' assignment 
- self._next_observation has been modified to output the entire window size of observations rather than just the last obs 

Exchange.py 
- observation_space is still here, I kept the functionality because I believe it could be potentially desirable behavior depending on the type of transform. Especially if the transform significantly changes the observation bounds, but this method will only be called once in the beginning, unlike before where transfrom_space was called on every obs. 

Simulated_exchange.py 
- data_frame indexed and set differently 
- observation columns are set at the beginning and then accessed via attribute rather than being calculated every time 
- self._next_observation modified to work with new talib indicator, padding is done before transform and index is not ignored (as the date is set to be the index), this is because it ensures a unique index when moving the obs through the indicators, which may have their own histories. 
- trade validation has been modified, trades that were "HOLDS" are still appended to the trades object in exchange, for rendering purposes 
- trade.step accessed when possible instead of self._current_step, this is to avoid wrong step issues
- performance is initialized with the initial balance and net worth
- net-worth now requires a step size to retrieve

Talib
Modified talib indicator for the following features: 
- Arguments are mapped to the indicator to avoid input errors via a parsing method  
- indicator results are calculated on a history of inputs, to avoid errors from too few values and a more accurate indicator 
-the final n values(self.n_window_size) from the history is then returned 

Render 
- TradingChart.py
- Dash sample files for development

### CODE REVIEW

**Comment by RaedShabbir** [comment]
The indicator must be initilized as follows
`ta_indicators = [('RSI', ['close']), 
                    ('MACD', ['close']), 
                    ('SAR', ['high','low']), 
                    # ('STOCH', ['high', 'low', 'close']), #outputs a tuple, can't work with that in current pipeline 
                    ('ADX', ['high', 'low', 'close']),
                    ('AD', ['high', 'low', 'close', 'volume']),
                    ('OBV', ['close', 'volume']),
                    ('BBANDS', ['close']), 
                    ('EMA', ['close']),
                    ('MA', ['close'])]
                    
    ta_indicators = TAlibIndicator(ta_indicators)`

File: tensortrade/features/indicators/talib_indicator.py [hunk 1]
```diff
@@ -21,28 +21,87 @@
...
+        #add new lows and highs for indicators in obs space
+        for i in range(len(self.hist_cols)):
+            new_lows = np.array([self._lows[i]] * self._window_size)[:,None]
+            new_highs = np.array([self._highs[i]] * self._window_size)[:,None]
+            output_space.low = np.concatenate((output_space.low, new_lows), axis=1)
+            output_space.high = np.concatenate((output_space.high, new_highs), axis=1)
+
+        return output_space
+
+    def transform(self, X: pd.DataFrame, input_space: Space) -> pd.DataFrame:
```
Comment (junfang2019) [inline tensortrade/features/indicators/talib_indicator.py]: in the new version Tensorflow, there is no need to use input_space parameters

**Comment by RaedShabbir** [comment]
Hey yeah, I just merged my repo yesterday, will upload an updated version
later today.

**Comment by RaedShabbir** [comment]
Input space parameters have been removed, transform_space method has been removed.

**Pull request merged**